src/widget/wlibrarytableview.h

```diff
@@ -47,20 +59,12 @@ class WLibraryTableView : public QTableView, public virtual LibraryView {
 
   protected:
     void focusInEvent(QFocusEvent* event) override;
-
-    void saveNoSearchVScrollBarPos();
-    void restoreNoSearchVScrollBarPos();
+    virtual QString getStateKey() const = 0;
 
   private:
-    void loadVScrollBarPosState();
-    void saveVScrollBarPosState();
-
     const UserSettingsPointer m_pConfig;
     const ConfigKey m_vScrollBarPosKey;
 
-    QMap<TrackModel*, int> m_vScrollBarPosValues;
-
-    // The position of the vertical scrollbar slider, eg. before a search is
-    // executed
-    int m_noSearchVScrollBarPos;
+    // saves scrollposition/selection/etc
+    QMap<QString, ModelState*> m_vModelState;
```

Comment:
Did you consider to use QCache? It is a FIFO Cache and should work exactly for our purpose.  

Please rename to something better like m_modelStateCache. 
We have no other prefix rule than p for pointer in Mixxx.

Reply:
I obviously did not do my research if some proper battery is included. Fixed.